app/requests_handlers.py

```diff
@@ -185,218 +121,163 @@ def request_code_check_session(request, fulfillment_type,
-            'page_title': 'Request a new access code'
+            'display_region': display_region,
+            'locale': locale,
+            'page_title': page_title,
+            'request_type': request_type,
+            'partial_name': 'request-' + request_type + '-code'
         }
 
 
-@requests_routes.view('/request-access-code/enter-address')
-class RequestCodeEnterAddressHHEN(RequestCodeCommon):
+@requests_routes.view(r'/' + View.valid_display_regions + '/request-' +
+                      RequestCodeCommon.valid_request_types + '-code/enter-address/')
+class RequestCodeEnterAddress(RequestCodeCommon):
     @aiohttp_jinja2.template('request-code-enter-address.html')
     async def get(self, request):
         self.setup_request(request)
-        self.log_entry(request, 'request-access-code/enter-address')
-        return {
-            'display_region': 'en',
-            'page_title': 'What is your postcode?'
-        }
-
-    @aiohttp_jinja2.template('request-code-enter-address.html')
-    async def post(self, request):
-        self.setup_request(request)
-        self.log_entry(request, 'request-access-code/enter-address')
-        data = await request.post()
-        await RequestCodeCommon.get_postcode(request, data, 'HH', 'EN',
-                                             'en')
+        request_type = request.match_info['request_type']
+        display_region = request.match_info['display_region']
 
+        if display_region == 'cy':
+            page_title = 'Beth yw eich cod post?'
+            locale = 'cy'
+        else:
+            page_title = 'What is your postcode?'
+            locale = 'en'
 
-@requests_routes.view('/gofyn-am-god-mynediad/nodi-cyfeiriad')
-class RequestCodeEnterAddressHHCY(RequestCodeCommon):
-    @aiohttp_jinja2.template('request-code-enter-address.html')
-    async def get(self, request):
-        self.setup_request(request)
-        self.log_entry(request, 'request-access-code/enter-address')
+        self.log_entry(request, display_region + '/request-' + request_type + '-code/enter-address')
         return {
-            'display_region': 'cy',
-            'locale': 'cy',
-            'page_title': 'Beth yw eich cod post?'
+            'display_region': display_region,
+            'locale': locale,
+            'page_title': page_title,
+            'request_type': request_type
         }
 
     @aiohttp_jinja2.template('request-code-enter-address.html')
     async def post(self, request):
         self.setup_request(request)
-        self.log_entry(request, 'request-access-code/enter-address')
-        data = await request.post()
-        await RequestCodeCommon.get_postcode(request, data, 'HH', 'CY',
-                                             'cy')
+        request_type = request.match_info['request_type']
+        display_region = request.match_info['display_region']
 
+        if display_region == 'cy':
+            page_title = 'Beth yw eich cod post?'
```

Comment:
The variable page_title is not used and fails flake8.

Reply:
Removed